---
format: 
  revealjs: 
    theme:  [default, style.scss]
    transition: fade
    slide-number: true
execute:
  echo: true 
  output: asis
editor: source
---


# Analyzing GENIE Genomic Data with {genieBPC} and {gnomeR} {background-color="#007CBA" style="text-align: center;"}

```{r}
#| echo: false
#| results: false
set.seed(20230515)

knitr::opts_chunk$set(echo = TRUE, results = 'asis')

library(tidyverse)
library(gtsummary)
library(synapser)
library(genieBPC)

# let's check if reg gt tables work now. if so, ditch this
knit_print.gtsummary <- function(x, ...) {
   gtsummary::as_gt(x) |>
     gt::as_raw_html()
}

knit_print.gt_tbl <- function(x, ...) {
  gt::as_raw_html(x)
} 


registerS3method("knit_print", "gtsummary", knit_print.gtsummary)
registerS3method("knit_print", "gt_tbl", knit_print.gt_tbl)

# fill for font awesome icons
fa_fill <- "#606060"

synLogin()
```

![](images/gnomeR-logo.png){width="250px"}

# Processing Data {background-color="#007CBA" style="text-align: center;"}

# {genieBPC} + {gnomeR} Pipeline

![](images/genie-gnomer-pipline.png)
## Case Study

::: {.panel-tabset}

### Code

```{r}
#| echo: false
#| results: false

nsclc_2 <- pull_data_synapse("NSCLC", "v2.0-public")

nsclc_2_ex <- create_analytic_cohort(nsclc_2$NSCLC_v2.0, stage_dx = "Stage IV",
                                 histology = "Adenocarcinoma")
```

:::


## Processing Data: Issues To Address

    
1) **Multiple Samples Per Patient**

2)  **Cohort Inclusion**
    -   Samples with no alterations may be dropped when pulling data

3)  **Data Formats & Gene Standards Often Inconsistent**
    -   Column names, data formats and gene names may differ between studies or even within studies!
    
4)  **Multi-Institutional Studies Use Several Gene Panels**
    -   Some samples may be sequenced using different panels therefore the non overlapping genes have to be annotated as missing


5)  **Missing Data in Mutation Status**

    
:::{.notes}
1)  **Research Samples vs. IMPACT Samples**
    -   Some samples may be sequenced outside MSK or using different panels.
    -   [Check panels and annotate NAs accordingly]{.emphasized}
2)  **Cohort Inclusion**
    -   Samples with no alterations may be dropped when pulling data
    -   [Use `samples` arg of `create_gene_binary()` and check data]{.emphasized}
3)  **Data Formats & Gene Standards Often Inconsistent**
    -   Column names, data formats and gene names may differ between studies or even within studies!
    -   [Reformat data as necessary and check gene aliases (`recode_aliases = TRUE`)]{.emphasized}
4)  **Missing Data in Mutation Status**
    -   Mutation Status is sometimes missing data.
    -   [Check Mutation Status column]{.emphasized}
    
:::


## Process Data: Select One Sample Per Patient

:::: {.columns}

::: {.column width="40%"}

- Patients can have many NGS values
- Setting criteria -> 1 sample
- If multiple match the criteria, random sample taken

:::

::: {.column width="60%"}


```{r}
#| echo: true
nsclc_samp <- select_unique_ngs(
  data_cohort = nsclc_2_ex$cohort_ngs,
  oncotree_code = "LUAD",
  sample_type = "Metastasis",
  min_max_time = "max"
)

```

:::

::::

## Process Data: Format Data in Analysis-ready Matrix

- Overview of `create_gene_binary()` structure

```{r, `code-line-numbers`="|2-4" }
#| output-location: column
#| echo: true
#| eval: false
gnomeR::create_gene_binary(
  samples = NULL
  mutation = NULL,
  cna = NULL,
  fusion = NULL) %>%
  head()
```

## Process Data: Ensure Samples with No Alterations Are Included



```{r, `code-line-numbers`="2" }
#| output-location: column
#| echo: true
#| eval: false

gnomeR::create_gene_binary(
  samples = 
  mutation = gnomeR::mutations,
  cna = gnomeR::cna,
  fusion = gnomeR::sv) %>%
  head()
```


## Process Data: Gene Aliases


Highlight Gene Alias and Gene NA panel annotation

```{r}

```


## Process Data: Multi-institutional Panels

Highlight Gene Alias and Gene NA panel annotation

```{r}

```

## Process Data: Missing Mutation Data & Data Filters

-Does this occur with GENIE data where there are NAs for variant classification /mutation type?

- we may not need this slide

