---
format: 
  revealjs: 
    theme:  [default, style.scss]
    transition: fade
    slide-number: true
execute:
  echo: true 
  output: asis
editor: source
---


# Analyzing GENIE Genomic Data with {genieBPC} and {gnomeR} {background-color="#007CBA" style="text-align: center;"}

```{r}
#| echo: false
#| results: false

knitr::opts_chunk$set(echo = TRUE, results = 'asis')

library(tidyverse)
library(gtsummary)

knit_print.gtsummary <- function(x, ...) {
   gtsummary::as_gt(x) |>
     gt::as_raw_html()
}

knit_print.gt_tbl <- function(x, ...) {
  gt::as_raw_html(x)
} 


registerS3method("knit_print", "gtsummary", knit_print.gtsummary)
registerS3method("knit_print", "gt_tbl", knit_print.gt_tbl)

# fill for font awesome icons
fa_fill <- "#606060"
```

![](images/gnomeR-logo.png){width="250px"}

# Processing Data {background-color="#007CBA" style="text-align: center;"}

# {genieBPC} + {gnomeR} Pipeline

![](images/genie-gnomer-pipline.png)

## Processing Data: Issues To Address

1)  **Research Samples vs. IMPACT Samples**
    -   Some samples may be sequenced outside MSK or using different panels. You can usually tell by the sample ID (if P-000XXX-T0X-IMX, it is likely IMPACT with IMX indicating panel)
2)  **Cohort Inclusion**
    -   Samples with no alterations may be dropped when pulling data
3)  **Data Formats & Gene Standards Often Inconsistent**
    -   Column names, data formats and gene names may differ between studies or even within studies!
4)  **Missing Data in Mutation Status**
    -   Mutation Status is sometimes missing data.
    
    

## Processing Data: Issues To Address

MAYBE THIS SLIDE GOES AT END as RECAP AFTER WE SHOW FUNCTIONS?


1)  **Research Samples vs. IMPACT Samples**
    -   Some samples may be sequenced outside MSK or using different panels.
    -   [Check panels and annotate NAs accordingly]{.emphasized}
2)  **Cohort Inclusion**
    -   Samples with no alterations may be dropped when pulling data
    -   [Use `samples` arg of `create_gene_binary()` and check data]{.emphasized}
3)  **Data Formats & Gene Standards Often Inconsistent**
    -   Column names, data formats and gene names may differ between studies or even within studies!
    -   [Reformat data as necessary and check gene aliases (`recode_aliases = TRUE`)]{.emphasized}
4)  **Missing Data in Mutation Status**
    -   Mutation Status is sometimes missing data.
    -   [Check Mutation Status column]{.emphasized}



## Process Data in Matrix

- show/highlight which args help data processing issues 

```{r}
#| output-location: "column"
#| echo: true
gnomeR::create_gene_binary(mutation = gnomeR::mutations) %>%
  head()
```

## Process Data in Matrix: Multi-institutional Panels

Highlight Gene Alias and Gene NA panel annotation

```{r}

```

# Analyzing Data {background-color="#007CBA" style="text-align: center;"}

## Analyzing Data: Issues To Address

1) **OncoKB Annotation**
    - Some studies may require oncoKB annotation of data
    
2) **Multiple Testing and False Positives**
    - Very low prevalence genes are often not very informative in standard IMPACT analyses. Choose a threshold (e.g. 1% or 5 %) _a prior_ to limit number of tests. 
    - Report q-value (adjusted for multiple testing)
    
3) **Analyzing Tumor Mutation Burden Across Samples**
    - Although TMB is always measured as number of mutations per coding area, depth of sequencing can make this measure inconsistent between studies. 
    - Usually ok within IMPACT studies but not always comparable with WES studies

4) **Survival Analysis Biases**

## Summarize Data with tbl_genomic()

```{r}

```

## Summarize Data with tbl_genomic()
May want to analyze on the gene level only
Use summarize_by_gene() first

```{r}

```

## Analyzing Data: Multiple Testing

- Because each hypothesis test has a probability of producing a type I error (falsely rejecting a null hypothesis that is true), performing a large number of hypothesis tests virtually guarantees the presence of type I errors among the findings.

- Firstly, depending on size of data and type of analysis, choose a prevalence threshold for genes to test

- [Report tests with adjustment]{.emphasized}- Recommended to use false discovery rate adjustment as opposed to FWER (eg. bonferroni) as it controls for a low proportion of false positives, instead of guarding against making any false positive conclusion at all

- Reference: Goeman JJ, Solari A. Multiple hypothesis testing in genomics. Stat Med. 2014 May 20;33(11):1946-78. doi: 10.1002/sim.6082. Epub 2014 Jan 8. PMID: 24399688.

## Helper Functions for Multiple Testing

- subset_by_frequency
- subset_by_panel 
- tbl_genomic()

