---
format: 
  revealjs: 
    theme:  [default, style.scss]
    transition: fade
    slide-number: true
execute:
  echo: true 
  output: asis
editor: source
---


# Analyzing GENIE Genomic Data with {genieBPC} and {gnomeR} {background-color="#007CBA" style="text-align: center;"}

```{r}
#| echo: false
#| results: false
set.seed(20230515)

knitr::opts_chunk$set(echo = TRUE, results = 'asis')

library(tidyverse)
library(gtsummary)
library(synapser)
library(genieBPC)

# let's check if reg gt tables work now. if so, ditch this
knit_print.gtsummary <- function(x, ...) {
   gtsummary::as_gt(x) |>
     gt::as_raw_html()
}

knit_print.gt_tbl <- function(x, ...) {
  gt::as_raw_html(x)
} 


registerS3method("knit_print", "gtsummary", knit_print.gtsummary)
registerS3method("knit_print", "gt_tbl", knit_print.gt_tbl)

# fill for font awesome icons
fa_fill <- "#606060"

synLogin()
```

![](images/gnomeR-logo.png){width="250px"}

# Processing Data {background-color="#007CBA" style="text-align: center;"}

# {genieBPC} + {gnomeR} Pipeline

![](images/genie-gnomer-pipline.png)
## Case Study

::: {.panel-tabset}

### Code

```{r}
#| echo: false
#| results: false

nsclc_2 <- pull_synapse_data("NSCLC", "v2.0-public")
cohort <- create_analytic_cohort(nsclc_2$NSCLC_v2.0, stage_dx = "Stage IV",
                                 histology = "Adenocarcinoma")
```

### Data

```{r}
samples <- 
```

:::


## Processing Data: Issues To Address

    
1) **Multiple Samples Per Patient**

2)  **Cohort Inclusion**
    -   Samples with no alterations may be dropped when pulling data

3)  **Data Formats & Gene Standards Often Inconsistent**
    -   Column names, data formats and gene names may differ between studies or even within studies!
    
4)  **Multi-Institutional Studies Use Several Gene Panels**
    -   Some samples may be sequenced using different panels therefore the non overlapping genes have to be annotated as missing


5)  **Missing Data in Mutation Status**
    -   Mutation Status is sometimes missing data.

    
:::{.notes}
1)  **Research Samples vs. IMPACT Samples**
    -   Some samples may be sequenced outside MSK or using different panels.
    -   [Check panels and annotate NAs accordingly]{.emphasized}
2)  **Cohort Inclusion**
    -   Samples with no alterations may be dropped when pulling data
    -   [Use `samples` arg of `create_gene_binary()` and check data]{.emphasized}
3)  **Data Formats & Gene Standards Often Inconsistent**
    -   Column names, data formats and gene names may differ between studies or even within studies!
    -   [Reformat data as necessary and check gene aliases (`recode_aliases = TRUE`)]{.emphasized}
4)  **Missing Data in Mutation Status**
    -   Mutation Status is sometimes missing data.
    -   [Check Mutation Status column]{.emphasized}
    
:::





## Process Data: Select One Sample Per Patient

:::: {.columns}

::: {.column width="40%"}
Patients can have many NGS values.
```{r}
#| echo: false
#| results: true

# select 5 samples and show how many NGS they have

```

:::

::: {.column width="60%"}


```{r}
#| echo: true
nsclc_one_ngs <- select_unique_ngs()

#
```

:::

::::



select_unique_ngs slide

## Process Data: Format Data in Analysis-ready Matrix

- show/highlight which args help data processing issues in the next few slides

```{r, `code-line-numbers`="|2|4" }
#| output-location: column
#| echo: true
#| eval: false
gnomeR::create_gene_binary(
  samples = 
  mutation = gnomeR::mutations,
  cna = gnomeR::cna,
  fusion = gnomeR::sv) %>%
  head()
```

## Process Data: Ensure Samples with No Alterations Are included

- samples arg

```{r, `code-line-numbers`="2" }
#| output-location: column
#| echo: true
#| eval: false

gnomeR::create_gene_binary(
  samples = 
  mutation = gnomeR::mutations,
  cna = gnomeR::cna,
  fusion = gnomeR::sv) %>%
  head()
```


## Process Data: Gene Aliases


Highlight Gene Alias and Gene NA panel annotation

```{r}

```


## Process Data: Multi-institutional Panels

Highlight Gene Alias and Gene NA panel annotation

```{r}

```

## Process Data: Missing Mutation Data & Data Filters

-Does this occur with GENIE data where there are NAs for variant classification /mutation type?

- we may not need this slide


# Analyzing Data {background-color="#007CBA" style="text-align: center;"}

# {genieBPC} + {gnomeR} Pipeline

![](images/genie-gnomer-pipline.png)
## Case Study??

Maybe use a case study as example. If so, add an analysis slide with clinical endpoint data

## Analyzing Data: Issues/Steps To Address

EDIT THIS DOWN AND ALSO ADD PATHWAYS 

1) **Summarizing on Alteration or Gene Levels** 

2) **Multiple Testing and False Positives**
    - Very low prevalence genes are often not very informative in standard IMPACT analyses. Choose a threshold (e.g. 1% or 5 %) _a prior_ to limit number of tests. 
    - Report q-value (adjusted for multiple testing)
    
2) **Limited Power To Detect With Sparse Alterations**  
- Pathways

MAKE THIS MORE GENERAL, Maybe add a slide on Z score - omit maybe 
3) **Analyzing Tumor Mutation Burden Across Samples**
    - Although TMB is always measured as number of mutations per coding area, depth of sequencing can make this measure inconsistent between studies. 
    - Usually ok within IMPACT studies but not always comparable with WES studies

4) **OncoKB Annotation**
    - Some studies may require oncoKB annotation of data
    
5) CNA segmentation data???? - Maybe 

## Analyzing Data: Summarize Data with tbl_genomic()

```{r}

```

## Analyzing Data: Summarize Data with tbl_genomic()

May want to analyze on the gene level only
Use summarize_by_gene() first

```{r}

```

## Analyzing Data: Multiple Testing

Ct=u t this down or omit

- Because each hypothesis test has a probability of producing a type I error (falsely rejecting a null hypothesis that is true), performing a large number of hypothesis tests virtually guarantees the presence of type I errors among the findings.

- Firstly, depending on size of data and type of analysis, choose a prevalence threshold for genes to test

- [Report tests with adjustment]{.emphasized}- Recommended to use false discovery rate adjustment as opposed to FWER (eg. bonferroni) as it controls for a low proportion of false positives, instead of guarding against making any false positive conclusion at all

- Reference: Goeman JJ, Solari A. Multiple hypothesis testing in genomics. Stat Med. 2014 May 20;33(11):1946-78. doi: 10.1002/sim.6082. Epub 2014 Jan 8. PMID: 24399688.

Keep this 
Helper Functions for Multiple Testing: 

- subset_by_frequency
- subset_by_panel 
- tbl_genomic()

## Analyzing Data: Pathways

## Analyzing Data: Data Visualization

- could also mention color palettes 
- CNA heatmap?

## Summary

## Misc
- oncoKB
- oncoprints, etc 
- CNA?
